debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java

```diff
@@ -112,6 +112,7 @@ public abstract class AbstractLogMinerEventProcessor<T extends Transaction> impl
     private Instant lastProcessedScnChangeTime = null;
     private Scn lastProcessedScn = Scn.NULL;
     private boolean sequenceUnavailable = false;
+    private HashMap<Long, TableId> objectIdToTableIdCache = new HashMap<>();
```

Comment:
```suggestion
    private Map<Long, TableId> objectIdToTableIdCache = new HashMap<>();
```

How large the cache can be? Do we want to use an unbounded one?

Reply:
I'm also curious if the cache needs to be reset for every call to processSCNs.  Can it be kept around for the life time of the processor.

Reply:
@jpechane, there are some corner cases where we could see the cache be anywhere between 1x and 2x the number of captured tables; however, within a single mining step, that would be extremely rare. This cache only applies when the Hybrid strategy is enabled; it's unused otherwise.

@jeremy-l-ford, that's a good question. My concern with leaving it around indefinitely is that when a `DROP TABLE` is executed, the entry in the cache won't be evicted. Now, Oracle doesn't reuse OBJECT_IDs, so it isn't like the entry itself has a logical issue at runtime, just that the entry would consume some memory unnecessarily.

Perhaps falling back on @jpechane's suggestion with a bounded cache would be the best of both worlds. We would be able to guard against corner cases where the cache could consume valuable memory unnecessarily and have a way to evict old entries.

Are you both in agreement?

Reply:
I think the bounded cache would work.  Maybe even making its size configurable/unbounded?  

Also curious what you would think about this information being owned by the OracleDatabaseSchema object.  It could then respond to schema events and drop a table/remove from the cache.

Reply:
For now, the cache will be capped at a size of `10` but can be controlled using `internal.object.id.cache.size`, which must be greater than `0`. I also moved the code to the `OracleDatabaseSchema` class and cleaned-up the code references, including some small refactors around iteration & attribute names.